library/core/src/convert/mod.rs

```diff
@@ -722,6 +722,7 @@ where
     ///
     /// That is, this conversion is whatever the implementation of
     /// <code>[From]&lt;T&gt; for U</code> chooses to do.
+    #[inline(always)]
```

Comment:
I don't think we want this -- inlining is bottom-up not top down (yes, this is unintuitive). This means that `U::from` will be inlined into this function before this function is be considered for inlining. If we take this `#[inline(always)]`, then that "consider" becomes a "yes", even if it would be a mistake.